README.md

```diff
@@ -96,61 +91,75 @@ our [official documentation](https://docs.conjur.org/Latest/en/Content/Developer
-`conjur_account`, and`cert_file`.
+#### Step 2. Define ConjurrcData
 
+ConjurrcData is a data class containing all the 'non-credential' connection details.
 ```
-# .conjurrc
----
-cert_file: /Users/someuser/conjur-server.pem
-conjur_account: someaccount
-conjur_url: https://conjur-server
+conjurrc_data = ConjurrcData(conjur_url=conjur_url,account=account,cert_file = None)
 ```
 
-#### Storing credentials
+* conjur_url - URL of the Conjur server
+* account - the organizational Conjur account name
+* cert_file - a path to the Conjur rootCA file. This is required if you are initializing the client in `SslVerificationMode.SELF_SIGN`
+  or `SslVerificationMode.CA_BUNDLE` mode
+
+#### Step 3. Storing credentials
 
-When using the SDK, you can keep credentials in the system's native credential store instead of
-passing them in during initialization of the Client. By default, the Client will favor saving
-credentials (login ID and password) to the system's credential store. If the detected credential
-store is not one we support or is not accessible, the credentials will be written to a configuration
-file, `.netrc`, in plaintext.
+The client retrieves credentials from a credential store called `CredentialStore` which inherits from `CredentialsStoreInterface`. This
+approach enables storing the credentials in a safe location, and providing the credentials to the client on demand.
 
-If written to the `.netrc`, it is strongly recommended that you log out when not using the Conjur
-CLI. This removes the credentials from the `.netrc` file.
+The SDK provide two implementations of `CredentialsStoreInterface`.
+1. `KeystoreCredentialsProvider` which saves credentials to the system's credential store (macOS keychain for example)
+2. `FileCredentialsProvider` which saves the credentials into `.netrc` file, in plaintext.
 
-The `.netrc` file or (`_netrc` for Windows environments) contains credentials needed to log in to
-the Conjur endpoint and should consist of 'machine', 'login', and 'password'.
 
-Note that if you choose to create this file yourself, ensure you follow least privilege, allowing
-only the user who has created the file to have read/write permissions on it (`chmod 700 .netrc`).
+We provide the user with `CredentialStoreFactory` which create such Credential stores.
+
+By default, the `CredentialStoreFactory` favors creating a `KeystoreCredentialsProvider`. If the SDK cannot access the 
+operating system's credential store, then `CredentialStoreFactory` will create `FileCredentialsProvider` instead
+
+If credentials are written to the `.netrc`, we strongly recommend deleting those credentials when not using the
+SDK. The file is located in the user's home directory, for example: `/Users/my_username` in macOS 
+or `C:\Users\my_username` in Windows
+
+The `.netrc` file or (`_netrc` for Windows environments) contains credentials needed to log in to the Conjur endpoint
+and should consist of 'machine', 'login', and 'password'.
 
 ```
 # .netrc / _netrc
```

Comment:
@mbenita-Cyberark (@InbalZilberman ) I think the must be an "example" format:
http://conjur-myorg.example.com"
(example.com is a legal usage used for all example urls - https://en.wikipedia.org/wiki/Special-use_domain_name)

Reply:
IMPORTANT ^^^

Reply:
Done